cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/CoordTypeCode.java

```diff
@@ -0,0 +1,125 @@
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.fits.slice;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+
+public enum CoordTypeCode {
+    // Spatial type codes.
+    RA("RA--", "deg", CoordType.SPATIAL),
+    DEC("DEC-", "deg", CoordType.SPATIAL),
+    GLON("GLON", "deg", CoordType.SPATIAL),
+    GLAT("GLAT", "deg", CoordType.SPATIAL),
+    ELON("ELON", "deg", CoordType.SPATIAL),
+    ELAT("ELAT", "deg", CoordType.SPATIAL),
+
+    // Spectral type codes.
+    FREQ("FREQ", "Hz", CoordType.SPECTRAL),
+    ENER("ENER", "J", CoordType.SPECTRAL),
+    WAVN("WAVN", "/m", CoordType.SPECTRAL),
+    VRAD("VRAD", "m s-1", CoordType.SPECTRAL),
+    WAVE("WAVE", "m", CoordType.SPECTRAL),
+    VOPT("VOPT", "m s-1", CoordType.SPECTRAL),
+    ZOPT("ZOPT", "", CoordType.SPECTRAL),
+    AWAV("AWAV", "m", CoordType.SPECTRAL),
+    VELO("VELO", "m s-1", CoordType.SPECTRAL),
+    BETA("BETA", "", CoordType.SPECTRAL);
+
+    private final String typeCodeString;
+    private final String defaultUnit;
+    private final CoordType coordType;
+
+
+    CoordTypeCode(final String typeCodeString, final String defaultUnit, final CoordType coordType) {
+        this.typeCodeString = typeCodeString;
+        this.defaultUnit = defaultUnit;
+        this.coordType = coordType;
+    }
+
+    public boolean isSpectral() {
+        return coordType == CoordType.SPECTRAL;
+    }
+
+    public String getDefaultUnit() {
+        return defaultUnit;
+    }
+
+    public static String getDefaultUnit(final String ctype) {
+        if (ctype == null) {
+            return null;
+        }
+
+        final CoordTypeCode matchedCoordTypeCode =
+                Arrays.stream(values()).filter(coordTypeCode -> ctype.toUpperCase(Locale.ROOT).substring(0, 4).startsWith(
+                        coordTypeCode.typeCodeString)).findFirst().orElse(null);
```

Comment:
Since this uses startsWith, you don't need the substring(0,4) here and in principle it could fail if the ctype is short which is technically valid (as noted in previous comment).